api/src/main/java/marquez/api/OpenLineageResource.java

```diff
@@ -65,13 +68,27 @@ public void create(
             (result, err) -> {
               if (err != null) {
                 log.error("Unexpected error while processing request", err);
-                asyncResponse.resume(Response.status(INTERNAL_SERVER_ERROR).build());
+                asyncResponse.resume(Response.status(determineStatusCode(err)).build());
               } else {
                 asyncResponse.resume(Response.status(201).build());
               }
             });
   }
 
+  private int determineStatusCode(Throwable e) {
+    if (e instanceof CompletionException) {
+      return determineStatusCode(e.getCause());
+    } else if (e instanceof IllegalArgumentException) {
+      return BAD_REQUEST.getStatusCode();
+    } else if (e instanceof MarquezHttpException) {
```

Comment:
Outside the integration tests, I don't think it's possible for an `MarquezHttpException` to be thrown. Do we need to handle the exception?